### PR TITLE
Renesas: Smartbond: Crypto Driver Update

### PR DESCRIPTION
**Pull request by ioannis-karachalios**
This PR should fix few Crypto driver issues and add support to the Crypto tests/samples suite.

### CODE REVIEW

File: drivers/crypto/crypto_smartbond.c [hunk 1]
```diff
@@ -253,13 +253,29 @@ static int crypto_smartbond_hash_set_out_len(void)
...
-		sys_mem_swap(data, sizeof(uint32_t));
-
-		return (*(uint32_t *)data);
+		/*
+		 * \warning Do not use in-memory swapping by calling \sa sys_mem_swap
+		 *          as the key might reside in the flash memory and thus any
+		 *          write access while XiP, should result in raising busfault
+		 *          exception.
+		 */
+		return crypto_swap_word(data);
```
Comment (ydamigos) [inline drivers/crypto/crypto_smartbond.c:278]:
```suggestion
		return SWAP32([__UNALIGNED_UINT32_READ(data));
```

Comment (ydamigos) [reply drivers/crypto/crypto_smartbond.c:278]: Or better use API functions like sys_get_le32()/sys_get_be32().

File: tests/crypto/crypto_hash/prj_mtls_shim.conf
Comment (ydamigos) [inline tests/crypto/crypto_hash/prj_mtls_shim.conf:1]: We need a "consumer" for this, don't we?

Comment (ioannis-karachalios) [reply tests/crypto/crypto_hash/prj_mtls_shim.conf:1]: Fixed

**Review by ydamigos — APPROVED** [review]

**Review by blauret — DISMISSED** [review]

**Review by andrzej-kaczmarek — DISMISSED** [review]

**Review by blauret — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Pull request merged**